pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go

```diff
@@ -1156,25 +1157,54 @@ type cutoverProvider interface {
 // custoverFromJobProgress is a cutoverProvider that decides whether the cutover
 // time has been reached based on the progress stored on the job record.
 type cutoverFromJobProgress struct {
-	registry *jobs.Registry
-	jobID    jobspb.JobID
+	db    isql.DB
+	jobID jobspb.JobID
+}
+
+func (c *cutoverFromJobProgress) loadIngestionProgress(
+	ctx context.Context,
+) (*jobspb.StreamIngestionProgress, error) {
+	var (
+		progressBytes []byte
+		exists        bool
+	)
+	if err := c.db.Txn(ctx, func(ctx context.Context, txn isql.Txn) error {
+		infoStorage := jobs.InfoStorageForJob(txn, c.jobID)
+		var err error
+		progressBytes, exists, err = infoStorage.GetLegacyProgress(ctx)
+		return err
+	}); err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, nil
+	}
+	progress := &jobspb.Progress{}
+	if err := protoutil.Unmarshal(progressBytes, progress); err != nil {
+		return nil, err
+	}
+
+	sp, ok := progress.GetDetails().(*jobspb.Progress_StreamIngest)
+	if !ok {
+		return nil, errors.Newf("unknown progress details type %T in stream ingestion job %d",
+			progress.GetDetails(), c.jobID)
+	}
+	return sp.StreamIngest, nil
 }
 
 func (c *cutoverFromJobProgress) cutoverReached(ctx context.Context) (bool, error) {
-	j, err := c.registry.LoadJob(ctx, c.jobID)
+	ingestionProgress, err := c.loadIngestionProgress(ctx)
 	if err != nil {
 		return false, err
 	}
-	progress := j.Progress()
-	var sp *jobspb.Progress_StreamIngest
-	var ok bool
-	if sp, ok = progress.GetDetails().(*jobspb.Progress_StreamIngest); !ok {
-		return false, errors.Newf("unknown progress type %T in stream ingestion job %d",
-			j.Progress().Progress, c.jobID)
+	if ingestionProgress == nil {
+		log.Warningf(ctx, "no legacy job progress recorded yet")
```

Comment:
when would this occur? Like before the job info table migration? If you don't know off the top of your head, I can do some digging.

Reply:
I would hope the only happen is during the migration which shouldn't really be that relevant for us.